ta/pkcs11/src/processing.c

```diff
@@ -440,13 +452,31 @@ enum pkcs11_rc entry_processing_step(struct pkcs11_client *client,
 	if (rc)
 		return rc;
 
+	if (step == PKCS11_FUNC_STEP_UPDATE_KEY) {
+		assert(function == PKCS11_FUNCTION_DIGEST);
+
+		rc = serialargs_get(&ctrlargs, &key_handle, sizeof(uint32_t));
+		if (rc)
+			return rc;
+	}
+
 	if (serialargs_remaining_bytes(&ctrlargs))
 		return PKCS11_CKR_ARGUMENTS_BAD;
 
 	rc = get_active_session(session, function);
 	if (rc)
 		return rc;
 
+	if (step == PKCS11_FUNC_STEP_UPDATE_KEY) {
+		assert(function == PKCS11_FUNCTION_DIGEST);
+
+		obj = pkcs11_handle2object(key_handle, session);
+		if (!obj) {
+			rc = PKCS11_CKR_KEY_HANDLE_INVALID;
+			goto out;
+		}
```

Comment:
Do we need a check to see, if object/key is accessible to this user i.e a call to check_access_attrs_against_token() ? I don't see an error code corresponding to PKCS11_CKR_USER_NOT_LOGGED_IN in the spec. But, i would think the same rules as any other object should apply.

Reply:
Isn't it enough if it is in `entry_processing_init()` ?

Reply:
Nope. Will add.

Reply:
Perhaps `CKR_KEY_HANDLE_INVALID` is OK in this case?

Reply:
Yes,  CKR_KEY_HANDLE_INVALID seems ok.